sdk/core/azure-core/azure/core/pipeline/policies/universal.py

```diff
@@ -96,7 +96,96 @@ def on_request(self, request, **kwargs):
+
+
+class TelemetryPolicy(SansIOHTTPPolicy):
+    """Telemetry policy for monitoring what SDK language, client library version,
+    and language/platform info a client is using to call into their service.
+
+    Clients can prepend additional information indicating the name and version of the client application.
+
+    :param str package_user_agent: The package information in the following format: <package_name>/<package_version>.
+        Example: "storage/4.0.0"
+
+    **Keyword arguments:**
+
+    *user_agent_overwrite (bool)* - Overwrites User-Agent when True. Defaults to False.
+
+    *user_agent_use_env (bool)* - Gets user-agent from environment. Defaults to True.
+    """
+
+    _USERAGENT = "User-Agent"
+    _ENV_ADDITIONAL_USER_AGENT = 'AZURE_HTTP_USER_AGENT'
+    _X_MS_AZSDK_Telemetry = "X-MS-AZSDK-Telemetry"
+
+    def __init__(self, package_user_agent=None, **kwargs):
+        # type: (Optional[str], Any) -> None
+        self.overwrite = kwargs.pop('user_agent_overwrite', False)
+        self.use_env = kwargs.pop('user_agent_use_env', True)
+        self.package_user_agent = package_user_agent
+
+        if package_user_agent is None:
+            self._user_agent = "azsdk-python-core/{} Python/{} ({})".format(
+                azcore_version,
+                platform.python_version(),
+                platform.platform(),
+            )
+        else:
+            self._user_agent = "azsdk-python-{} Python/{} ({})".format(
+                package_user_agent,
+                platform.python_version(),
+                platform.platform(),
+            )
+
+    @property
+    def user_agent(self):
+        # type: () -> str
+        """The current user agent value."""
+        if self.use_env:
+            add_user_agent_header = os.environ.get(self._ENV_ADDITIONAL_USER_AGENT, None)
+            if add_user_agent_header is not None:
+                return "{} {}".format(self._user_agent, add_user_agent_header)
+        return self._user_agent
+
+    def add_user_agent(self, value):
+        # type: (str) -> None
+        """Add value to current user agent with a space.
+
+        :param str value: value to add to user agent.
+        """
+        self._user_agent = "{} {}".format(self._user_agent, value)
+
+    def on_request(self, request, **kwargs):
+        # type: (PipelineRequest, Any) -> None
+        """Modifies the User-Agent header before the request is sent.
+
+        :param request: The PipelineRequest object
+        :type request: ~azure.core.pipeline.PipelineRequest
+        """
+        telemetry_disabled = os.environ.get('AZURE_TELEMETRY_DISABLED', False)
+        if telemetry_disabled:
+            return
+
+        http_request = request.http_request
+        options = request.context.options # type: ignore
+        existing = request.http_request.headers.get(self._USERAGENT, "") # type: ignore
+
+        if existing:
+            self._user_agent = "{} {}".format(self.user_agent, existing)
```

Comment:
Isn't this what `self.add_user_agent(existing)` would do?

Reply:
Actually I guess `add_user_agent` would append to `self._user_agent` (which does not contain any env added content) but this appends to `self.user_agent` (which might contain env added content) Which actually raises an question: if `existing` is True, and `add_user_agent_header` is also True, won't subsequent references to `self.user_agent` have two copies of `add_user_agent_header` in them?

Reply:
You are correct - the old logic did create duplicate copies of the env user-agent header. I added a check to prevent this.